### PR TITLE
Parser update and publish

### LINKED ISSUE #21: Update Course parser to take section/workout list
Currently the `Course` parser reads a list of sections, and then looks at the workouts in the folder and orders the workouts by their `parent` tag. This is a pretty fragile way to do it and it's unintuitive to use.

Let's make each Course README responsible for knowing what workouts go in what order (and what section they're in.)

Example, for the JavaScript Core Course:
```
sections:
  '0':
    - intro-js
    - syntax
    - types
    - flow-control
    - functions-i
    - arrays
    - objects
  '1':
    - strings
    - array-methods
    - functions-ii
    - scope
    - errors
    - types-ii
    - numbers
  '2':
    - functions-iii
    - recipes-i
    - in-depth-ii
    - recipes-ii
  '3':
    - async
    - async-tips
```

### LINKED ISSUE #22: create `toJSON` methods for each parser
For each parser, add a `toJSON` method that outputs the properties in a format such that they can update the database. Basically, look at our mongoose models and make sure that we can use the output of this function to create or update a mongoose model.

- [ ] Add toJSON to `Insight`
- [ ] Add toJSON to `Workout`
- [ ] Add toJSON to `Course`
- [ ] Add toJSON to `Topic`

### PR DESCRIPTION
**Pull request by Loopiezlol**
- [x] added Mihai's fixes to the constructors from https://github.com/sagelabs/curriculum-tools/pull/20
- [x] added `.toJSON()` function to `Insight`, `Workout`, `Topic`, `Course`, to be used by the monoenki's content consumer
- [x] made it an npm package
- [x] parse `game:` field in workouts
- [x] parse `next/prereq` fields in Course 
```
next: [ {topicSlug, subtopicSlug }]
```

closes #22 
closes #21

### CODE REVIEW

**Comment by mihaiberq** [comment]
:+1:

File: lib/course.js [hunk 1]
```diff
@@ -1,80 +1,113 @@
...
+   * @param {string} Course readme text
+   */
   parse(text) {
-    if (text.length == 0) throw new Error("Passed in empty readme text");
-    let parsed = yaml.safeLoad(text);
-    if (parsed.sections) {
-      parsed.sections.forEach((s) => {this.sections[s] = []});
-      if (parsed.sections.workouts) parsed.sections.forEach((s) => {s.workouts.forEach((w) => {this.sections[s].push(w)})})
+    if (text.length === 0) {
+      throw new Error('Passed empty Topic README');
```
Comment (mihaiberq) [inline lib/course.js]: Topic -> Course

File: lib/course.js [hunk 2]
```diff
@@ -1,80 +1,113 @@
...
+    }
+    yaml.safeLoadAll(text.split('---')[0], doc => {
```
Comment (mihaiberq) [inline lib/course.js]: Don't think splitting here is needed

File: lib/course.js [hunk 3]
```diff
@@ -1,80 +1,113 @@
...
+      }
+    });
+
+    if (this.next) {
+      this.next = this.next.map(entry => {
+        const [ topicSlug, subtopicSlug ] = entry.split(':');
+
+        return {
+          topic: topicSlug,
+          subtopic: subtopicSlug
```
Comment (mihaiberq) [inline lib/course.js]: does the db schema use "subtopic" still?

File: lib/course.js [hunk 4]
```diff
@@ -1,80 +1,113 @@
...
+      });
+    }
+
+    if (this.prerequisites) {
+      this.prerequisites = this.prerequisites.map(entry => {
+        const [ topicSlug, subtopicSlug ] = entry.split(':');
+
+        return {
+          topic: topicSlug,
+          subtopic: subtopicSlug
```
Comment (mihaiberq) [inline lib/course.js]: same here

File: lib/course.js (hunk 3 above)
Comment (Loopiezlol) [reply lib/course.js]: yeah our db models still use `subtopic`

File: lib/course.js (hunk 4 above)
Comment (Loopiezlol) [reply lib/course.js]: same as above

File: lib/course.js (hunk 2 above)
Comment (Loopiezlol) [reply lib/course.js]: true, merci

File: lib/course.js (hunk 1 above)
Comment (Loopiezlol) [reply lib/course.js]: nice catch

**Pull request merged**